src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -272,6 +309,15 @@ public void update(Tangle tangle, Snapshot initialSnapshot, String item) throws
         if (initialSnapshot.hasSolidEntryPoint(hash)) {
             return;
         }
+
+        TransactionViewModel cachedTvm = tangle.getCache(TransactionViewModel.class).get(hash);
+        if (cachedTvm != null) {
+            this.isCacheEntryFresh = false;
```

Comment:
what about `dirty`?

Btw, I would say that currently it is fine that `fresh` and `dirty` is part of ViewModel.
Maybe in a later PR (not now) it will make sense to have a `ViewModel` abstract class and interface to hold those values...
It is kind of annoying because we want to get rid of ViewModel but not much choice

Reply:
I'm using `isCacheEntryFresh` boolean to store fresh or dirty state.